### PR TITLE
Added rule for type inference

### PR DESCRIPTION
**Pull request by Haud**
@paulmiard @ThibaultKlein

### CODE REVIEW

**Comment by ThibaultKlein** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -316,6 +317,31 @@ In all use-cases, the colon should be associated with the left-most item with no
...
 ```
 
+### Type Inference ###
+
+Prefer letting the compiler infer the type instead of explcitly stating it, wherever possible:
+
+```swift
+var max = 0 		// Int
+var name = "John" 	// String
+var rect = CGRect() // CGRect
```
Comment (paulmiard) [inline README.md]: align comments?

**Comment by paulmiard** [comment]
:+1:

File: README.md (hunk 1 above)
Comment (paulmiard) [inline README.md]: this is not supposed to be here. Xcode got confused. Maybe when you rebase it will refer to the right line again?

Comment (Haud) [inline README.md]: Done

**Comment by Haud** [comment]
@paulmiard I updated your comments. I know you +1'd, but can you just run through one last time and make sure it's all good before merging?

**Comment by Haud** [comment]
@paulmiard bump

File: README.md [hunk 2]
```diff
@@ -12,6 +12,7 @@
 	* [Force Unwrap](#force-unwrap)
 	* [Access Modifiers](#access-modifiers)
 	* [Type Declarations](#type-declarations)
+	* [Type Inference](#type-inference)
```
Comment (paulmiard) [inline README.md:15]: Github won't let me comment on something not in this PR but Standards doesn't have its own link. It's bothering me

**Comment by paulmiard** [comment]
@Haud yup good to go. I made a comment that might not be related but also doesn't deserve its own PR. If you feel like fixing go for it otherwise you can just merge

**Comment by Haud** [comment]
@paulmiard File an issue and I can take care of it in a subsequent PR.

**Pull request merged**